backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1758,3 +1758,52 @@ databaseChangeLog:
             columns:
               - column:
                   name: test_result_delivery
+  - changeSet:
+      id: add-patient-registration-link-table
+      author: nicholas.a.scialli@omb.eop.gov
+      comment: Add a table that stores facility- or organization-specific patient registration links.
+      changes:
+        - createTable:
+            tableName: patient_registration_link
+            remarks: This table stores unique registration links associated with a facility or organization.
+            columns:
+              - column: *pk_column
+              - column: *created_at_column
+              - column: *created_by_column
+              - column: *updated_at_column
+              - column: *updated_by_column
+              - column:
+                  name: patient_registration_link
+                  remarks: The unique link
+                  type: *string
+                  constraints:
+                    nullable: false
+                    unique: true
+              - column:
+                  name: facility_id
+                  type: *idtype
+                  remarks: The facility associated with the link
+                  constraints:
+                    nullable: true
+                    unique: true
+                    foreignKeyName: fk__link__facility
+                    references: facility
+              - column:
+                  name: organization_id
+                  type: *idtype
+                  remarks: The organization associated with the link
+                  constraints:
+                    nullable: true
+                    unique: true
```

Comment:
Definitely not true.

Reply:
I believe the way I have designed this you'd never have an organization ID in this table more than once. If the `patient_registration_link` maps to a facility, that row would only have a `facility_id` and if the `patient_registration_link` maps to an organization, that row would only have an `organization_id`. So I _think_ the `unique` constraint holds?

Reply:
I don't see what's broken about this either...?

Reply:
Sorry, I lost track of this conversation.

It seems like I'm confused about the expected usage pattern here—my initial assumption was that these were expirable links, but it seems like they're a durable optional property of the organization or facility, which means I'm wrong about the uniqueness. (There are exciting and unpleasant things to discover about unique indexes and nullable fields, but I think this pattern mostly avoids them, with the exception of some possible minor performance issues that don't matter for this scale.)

I'm still not _not_ confused about the purpose of the entity, but I agree that I was mistaken about the constraint being wrong, and should have responded more swiftly to the correction. Sorry!